core/streamdata.go

```diff
@@ -76,10 +76,20 @@ func NetSegData(md *SegTranscodingMetadata) (*net.SegData, error) {
 			allTS = false
 		}
 	}
-	if allTS {
-		segData.FullProfiles = fullProfiles
+	allIntFPS := true
+	for i := 0; i < len(md.Profiles) && allIntFPS; i++ {
+		if md.Profiles[i].FramerateDen != 0 {
+			allIntFPS = false
+		}
+	}
+	if allIntFPS {
+		if allTS {
+			segData.FullProfiles = fullProfiles
+		} else {
+			segData.FullProfiles2 = fullProfiles
+		}
 	} else {
-		segData.FullProfiles2 = fullProfiles
+		segData.FullProfiles3 = fullProfiles
```

Comment:
Needs test cases for generating FullProfiles3 with frac FPS. This behavior has been mostly tested within [`TestGenSegCreds_FullProfiles`](https://github.com/livepeer/go-livepeer/blob/d9ea3358e0a43f07a0fd78880f37bd48750c1be2/server/segment_rpc_test.go#L225) and I believe that will also ensure that the back-compat behavior is preserved.

(That particular test case is currently a bit out of place ever since NetSegData was factored out of `genSegCreds` but I forgot to also move out the test cases as part of https://github.com/livepeer/go-livepeer/pull/1511 . Feel free to put the new test wherever you like.)